android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/BeaconConfigFragment.java

```diff
@@ -172,6 +174,17 @@ public void onPause() {
   }
 
   @Override
+  public void onDestroy() {
+    super.onDestroy();
+    if (mShortenUrlTask != null) {
```

Comment:
Once you move it there what happens if you try to cancel an already canceled task? You might need to set `mShortenUrlTask = null` to avoid that.